sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -305,6 +306,63 @@ case class HashPartitioning(expressions: Seq[Expression], numPartitions: Int)
     newChildren: IndexedSeq[Expression]): HashPartitioning = copy(expressions = newChildren)
 }
 
+/**
+ * Represents a partitioning where rows are split across partitions based on transforms defined
+ * by `expressions`. `partitionValuesOpt`, if defined, should contain value of partition key(s) in
+ * ascending order, after evaluated by the transforms in `expressions`, for each input partition.
```

Comment:
> ascending order

I thought it's ordered by partition id, e.g. `[part1_value, part2_value, ...]`. Are we require the partitioning to sort its partitions by the partition value in ascending order?

Reply:
If that's the case, we should document it in the public v2 class `KeyGroupedPartitioning`

Reply:
nvm, Spark groups partitions and sorts the partitions values by itself, in `BatchScanExec`